File: testdata/jennies/builders/constructor_initializations/PythonBuilder/builders/constructor_initializations.py
import typing
from ..cog import builder as cogbuilder
from ..models import constructor_initializations


class SomePanel(cogbuilder.Builder[constructor_initializations.SomePanel]):
    _internal: constructor_initializations.SomePanel

    def __init__(self):
        self._internal = constructor_initializations.SomePanel()        
        self._internal.type_val = "panel_type"        
        self._internal.cursor = constructor_initializations.CursorMode.TOOLTIP

    def build(self) -> constructor_initializations.SomePanel:
        """
        Builds the object.
        """
        return self._internal    
    
    def title(self, title: str) -> typing.Self:    
        self._internal.title = title
    
        return self
    

File: testdata/jennies/builders/constructor_initializations/RustBuilder/src/builders/constructor_initializations.rs
use crate::cog;
use crate::types::constructor_initializations;

#[derive(Debug, Clone)]
pub struct SomePanelBuilder {
    internal: constructor_initializations::SomePanel,
    pub(crate) errors: Vec<cog::BuildError>,
}

impl SomePanelBuilder {
    pub fn new() -> Self {
        let mut builder = Self {
            internal: constructor_initializations::SomePanel::default(),
            errors: Vec::new(),
        };
        builder.internal.r#type = "panel_type".to_string();
        builder.internal.cursor = constructor_initializations::CursorMode::Tooltip;

        builder
    }
}

impl Default for SomePanelBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl SomePanelBuilder {
    pub fn title(mut self, title: String) -> Self {
        self.internal.title = title;

        self
    }
}

impl cog::Builder<constructor_initializations::SomePanel> for SomePanelBuilder {
    fn build(&self) -> Result<constructor_initializations::SomePanel, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(self.internal.clone())
    }
}
